### PR TITLE
Accept IValueDescriptor in ModelBinder

### PR DESCRIPTION
**Pull request by couven92**
Currently the `BindMemberFrom...` methods on `ModelBinder` and `ModelBinder<T>` currently only accept `Option` or `IOption`. Thus, binding an `Argument` will only work if the target member matches the argument name. `BindMemberFrom...` can be used to configure the binder to bind any option to any member regardless of name matching.

There is no real reason why you should not be able to use `BindMemberFrom...` with an `Argument` or `IArgument`.

* Renames internal class `OptionValueSource` to `SpecificSymbolValueSource` (matching the filename the type is located in)  
   Changes its `TryGetValue` implementation to use `CurrentSymbolValueSource` with value descriptor provided at construction.
* **Renames publically visible method `BindMemberFromOption` to `BindMemberFromValue`**
* Changes argument in `BindMemberFrom...` from `Option` and `IOption` to `IValueDescriptor`

As this will introduce a breaking change, we can also choose to not rename the public method `BindMemberFromOption`, add the new method, and forward the call from the old one to the new more general one.

### CODE REVIEW

File: src/System.CommandLine.Tests/Binding/ModelBinderTests.cs [hunk 1]
```diff
@@ -319,7 +319,7 @@ public void Values_from_options_on_parent_commands_can_be_bound_regardless_of_na
 
             var binder = new ModelBinder<ClassWithMultiLetterSetters>();
 
-            binder.BindMemberFromOption(
+            binder.BindMemberFromValue(
```
Comment (jonsequitur) [inline src/System.CommandLine.Tests/Binding/ModelBinderTests.cs]: It's probably worthwhile to also add a test for binding from an `Argument`.

**Comment by couven92** [comment]
Hmm... it seems the Beast finally cought up with me... Appearently the ModelBinder does not really like binding against Arguments. I am investigating how to make it behave

**Comment by couven92** [comment]
@jonsequitur Can you look at the build? I have no idea why it fails. It claims that I do not have access to `ArgumentResult2`, which is an internal type within the same assembly and namespace. And it does not help that it build fine on my machine...

File: src/System.CommandLine/Binding/SpecificSymbolValueSource.cs [hunk 2]
```diff
@@ -1,34 +1,47 @@
...
-            IValueDescriptor valueDescriptor, 
-            BindingContext bindingContext, 
+        public bool TryGetValue(IValueDescriptor valueDescriptor,
+            BindingContext bindingContext,
             out object boundValue)
         {
-            var result = bindingContext.ParseResult.FindResultFor(Option);
-
-            switch (result)
+            valueDescriptor = ValueDescriptor;
```
Comment (Keboo) [inline src/System.CommandLine/Binding/SpecificSymbolValueSource.cs]: This always overwrites the passed in parameter with the property. Did you mean `valueDescriptor ??= ValueDescriptor;`?

File: src/System.CommandLine/ParseResult.cs [hunk 3]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Collections.Generic;
+using System.CommandLine.Binding;
```
Comment (Keboo) [inline src/System.CommandLine/ParseResult.cs]: Is this change needed?

File: src/System.CommandLine/Binding/SpecificSymbolValueSource.cs [hunk 4]
```diff
@@ -1,34 +1,47 @@
...
 namespace System.CommandLine.Binding
 {
-    internal class OptionValueSource : IValueSource
+    internal class SpecificSymbolValueSource : IValueSource
     {
-        public OptionValueSource(IOption option)
+        public SpecificSymbolValueSource(IValueDescriptor valueDescriptor)
         {
-            Option = option;
+            ValueDescriptor = valueDescriptor;
```
Comment (Keboo) [inline src/System.CommandLine/Binding/SpecificSymbolValueSource.cs]:
```suggestion
            ValueDescriptor = valueDescriptor ?? throw new ArgumentNullException(nameof(valueDescriptor));
```

File: src/System.CommandLine/ParseResult.cs (hunk 3 above)
Comment (couven92) [reply src/System.CommandLine/ParseResult.cs]: nope

File: src/System.CommandLine/Binding/SpecificSymbolValueSource.cs (hunk 2 above)
Comment (couven92) [reply src/System.CommandLine/Binding/SpecificSymbolValueSource.cs]: No, But I see that the code is misleading. I updated the implementation and during that also simplified it 😃
